### PR TITLE
Form validation and sanitisation done, along with insertion into data…

### PR DESCRIPTION
**Pull request by cwoodwarduk**
…base in a single function. Tests to be written.

### CODE REVIEW

File: upload.php [hunk 1]
```diff
@@ -1 +1,20 @@
 <?php
+
+if ( $_SERVER['REQUEST_METHOD']=='GET' && realpath(__FILE__) == realpath( $_SERVER['SCRIPT_FILENAME'] ) ) {
+    die ('<h1>Nice try, Charlie!</h1><h2>Try going to the form page here: </h2><br> <div class=\"buttonContainer\"> <a href="formPage.php"><button type=\"button\">GO BACK TO FORM</button></a></div>');
+}
+
+require_once 'functions.php';
+
+$connection = dbConn();
```
Comment (LukeChaston) [inline upload.php:9]: Did you not need the DB connection at the top of the page in the PDO doc format? I think Charlie said we needed it in that style.

File: formPage.php [hunk 2]
```diff
@@ -58,4 +64,4 @@
     </div>
```
Comment (LukeChaston) [inline formPage.php:64]: Does this Div closer actually close anything? I've looked at it a couple of times and can't see what it's for.

Comment (cwoodwarduk) [reply formPage.php:64]: Checked the div tags, there's no strays in there.

File: upload.php (hunk 1 above)
Comment (cwoodwarduk) [reply upload.php:9]: Just got the all-clear on that.

**Pull request merged**